packages/security/src/CredentialManagerOverride.ts

```diff
@@ -0,0 +1,254 @@
+     */
+    public static getCredMgrInfoByPlugin(pluginName: string) : ICredentialManagerNameMap | null {
+        let credMgrInfo: ICredentialManagerNameMap = null;
+        for (const nextNameMap of this.KNOWN_CRED_MGRS) {
+            if (nextNameMap.credMgrPluginName === pluginName) {
+                credMgrInfo = nextNameMap;
+                break;
+            }
+        }
+        return credMgrInfo;
+    }
+
+    //________________________________________________________________________
+    /**
+     * Get the credential manager information for the specified plugin.
+     *
+     * @param ZEExtName - Name of the Zowe Explorer extension
+     *
+     * @returns An ICredentialManagerNameMap or
+     *          null if the specified extension is not a known credential manager.
+     */
+    public static getCredMgrInfoByZEExt(ZEExtName: string) : ICredentialManagerNameMap | null {
+        let credMgrInfo: ICredentialManagerNameMap = null;
+        for (const nextNameMap of this.KNOWN_CRED_MGRS) {
+            if (nextNameMap.credMgrZEName === ZEExtName) {
+                credMgrInfo = nextNameMap;
+                break;
+            }
+        }
+        return credMgrInfo;
+    }
+
+    //________________________________________________________________________
+    /**
+     * Get the known credential managers.
+     *
+     * @returns An array of credential managers.
+     */
+    public static getKnownCredMgrs() : ICredentialManagerNameMap[] {
+        return this.KNOWN_CRED_MGRS;
+    }
+
+    /**
+     * Override the current credential manager with the specified
+     * credential manager. A plugin or ZE extension that provides a
+     * credential manager would override the default credential manager
+     * upon installation.
+     *
+     * @param newCredMgrName
+     *        The display name of your credential manager.
+     *
+     * @throws An ImperativeError upon error.
+     */
+    public static overrideCredMgr(newCredMgrName: string) : void {
+        const credMgrInfo: ICredentialManagerNameMap =
+            CredentialManagerOverride.getCredMgrInfoByDisplayName(newCredMgrName);
+        if (credMgrInfo === null) {
+            // We do not have a known credMgr. Form a list of known credential managers.
+            const knownCredMgrs: ICredentialManagerNameMap[] = CredentialManagerOverride.getKnownCredMgrs();
+            let knownMgrsMsg: string = "";
+            for ( let credMgrInx = 0; credMgrInx < knownCredMgrs.length; credMgrInx++) {
+                knownMgrsMsg += `\n${knownCredMgrs[credMgrInx].credMgrDisplayName}`;
+            }
+
+            // we do not permit overriding by an unknown credMgr
+            throw new ImperativeError({
+                msg: `The credential manager name '${newCredMgrName}' is an unknown ` +
+                `credential manager. The previous credential manager will NOT be overridden. ` +
+                `Valid credential managers are:` + knownMgrsMsg
```

Comment:
i like that you're returning the currently valid and active credential manager here. maybe include this type of reminder in some of the messages below too?

Reply:
Can you provide specific cases?

Reply:
I checked back and i think your updates covered my suggestion, I might have also just misread when commenting initially.